### PR TITLE
Type conversion improvements

### PR DESCRIPTION
**Pull request by lkorenc**
* Move type converters into their own header subdirectory.
* Properly convert data layouts.

### CODE REVIEW

File: include/vast/Conversion/TypeConverters/DataLayout.hpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    // LLVM types should have `abi:pref` where `pref` is not required, but we
+    // are going to emit it anyways to make things easier.
+    // This whole functionality is 100% dependent on internals of `LLVM::` dialect
+    // that are not exposed anywhere and are probably free to change on version bumps.
+    struct llvm_dl_entry_helper
+    {
+        static mlir_attr make(mcontext_t &mctx, mlir_type llvm_type,
+                              const dl::DLEntry &old_entry)
+        {
+            // TODO(conv:tc): This has more complicated rules, consult `LLVM` dialect
```
Comment (xlauko) [inline include/vast/Conversion/TypeConverters/DataLayout.hpp:80]: Make issue and link it in the comment

File: include/vast/Conversion/TypeConverters/DataLayout.hpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        static mlir_attr make(mcontext_t &mctx, mlir_type vast_type,
+                              const dl::DLEntry &old_entry)
+        {
+            auto new_entry = old_entry;
+            new_entry.type = vast_type;
+            return new_entry.create_raw_attr(mctx);
+        }
+    };
+
+    struct bultin_dl_entry_helper
```
Comment (xlauko) [inline include/vast/Conversion/TypeConverters/DataLayout.hpp:108]: typo `bultin` -> `builtin`

File: include/vast/Conversion/TypeConverters/DataLayout.hpp [hunk 3]
```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2021-present, Trail of Bits, Inc.
```
[the remark this answers is not in the document]
Comment (xlauko) [reply include/vast/Conversion/TypeConverters/DataLayout.hpp:1]: And other docs too. You have started using completely new indentation for `{` `}` in comparison to the rest of vast.

**Review by xlauko — APPROVED** [review]

File: include/vast/Conversion/TypeConverters/DataLayout.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (lkorenc) [reply include/vast/Conversion/TypeConverters/DataLayout.hpp:1]: Sorry for this, kinda lost myself.

**Pull request merged**